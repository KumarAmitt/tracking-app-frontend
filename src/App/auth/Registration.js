import React, { useState } from 'react';
import { Link, Redirect } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { getRegistrationStatus, registerUser } from '../../store/slicers/user_registration';
import AppBar from '../components/AppBar';

const Registration = () => {
  const dispatch = useDispatch();
  const registrationStatus = useSelector(getRegistrationStatus);

  const [newUser, setNewUser] = useState({ username: '', password: '', password_confirmation: '' });

  const handleChange = (e) => {
    setNewUser({ ...newUser, [e.target.name]: e.target.value });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(registerUser({
      user: {
        username: newUser.username,
        password: newUser.password,
        password_confirmation: newUser.password_confirmation,
      },
    }));
  };

  if (registrationStatus === true) {
    return <Redirect to="/dashboard" />;
  }

  return (
    <>
      <AppBar title="Sign up" />
      <form onSubmit={handleSubmit}>
        <input type="text" name="username" placeholder="Username" value={newUser.username} onChange={handleChange} required />
        <input type="password" name="password" placeholder="Password" value={newUser.password} onChange={handleChange} required />
        <input type="password" name="password_confirmation" placeholder="Password Confirmation" value={newUser.password_confirmation} onChange={handleChange} required />
        <button type="submit">Register</button>
      </form>
      <br />
      <p>Existing users Login</p>
      <Link to="/">Sign In</Link>
    </>
  );
};

export default Registration;
