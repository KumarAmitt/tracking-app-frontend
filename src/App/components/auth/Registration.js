import React, { useState } from 'react';
import { Link, Redirect } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import LinearProgress from '@material-ui/core/LinearProgress';
import AppBar from '../AppBar/AppBar';
import './style/auth.css';
import { getUserInfo, getUserLoadingStatus, registerUser } from '../../../store/slicers/user';

const Registration = () => {
  const dispatch = useDispatch();

  const userInfo = useSelector(getUserInfo);
  const userLoading = useSelector(getUserLoadingStatus);

  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');

  const resetForm = () => {
    setUsername('');
    setPassword('');
    setConfirmPassword('');
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    dispatch(registerUser({
      user: {
        username,
        password,
        password_confirmation: confirmPassword,
      },
    }));
    resetForm();
  };

  if (userInfo.logged_in) {
    return <Redirect to="/profile" />;
  }

  return (
    <>
      <AppBar title="Sign up" link="/" />
      <div className="auth">
        <div className="auth-sub">
          <p>Already have an account?</p>
          <Link to="/">Sign In</Link>
        </div>
        <form onSubmit={handleSubmit} id="form">
          <input type="text" placeholder="Username (4-8 character)" value={username} onChange={(e) => setUsername(e.target.value)} required />
          <input type="password" placeholder="Password" value={password} onChange={(e) => setPassword(e.target.value)} required />
          <input type="password" placeholder="Password Confirmation" value={confirmPassword} onChange={(e) => setConfirmPassword(e.target.value)} required />
          <button type="submit">Register</button>
        </form>
        {
          userLoading && <LinearProgress className="progress-bar" />
         }
      </div>
    </>
  );
};

export default Registration;
