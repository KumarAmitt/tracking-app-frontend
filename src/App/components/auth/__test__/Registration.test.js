import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureAppStore from '../../../../store/configureStore';
import Registration from '../Registration';

const store = configureAppStore();

describe('Registration', () => {
  it('should match the snapshot of Registration component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Registration />
        </BrowserRouter>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
