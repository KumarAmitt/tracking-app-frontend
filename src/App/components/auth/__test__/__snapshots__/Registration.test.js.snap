// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration should match the snapshot of Registration component 1`] = `
Array [
  <div
    className="appbar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
        />
      </svg>
    </a>
    <div
      className="title"
    >
      Sign up
    </div>
  </div>,
  <div
    className="auth"
  >
    <form
      id="form"
      onSubmit={[Function]}
    >
      <input
        name="username"
        onChange={[Function]}
        placeholder="Username (4-8 character)"
        required={true}
        type="text"
        value=""
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
      <input
        name="password_confirmation"
        onChange={[Function]}
        placeholder="Password Confirmation"
        required={true}
        type="password"
        value=""
      />
      <button
        type="submit"
      >
        Register
      </button>
    </form>
    <div
      className="auth-error hide"
    >
      Registration Failed
    </div>
    <div
      className="auth-sub"
    >
      <p>
        Existing users Login
      </p>
      <a
        href="/"
        onClick={[Function]}
      >
        Sign In
      </a>
    </div>
  </div>,
]
`;
